src/device/x86-linux/gpioctl_x86_linux.cpp

```diff
@@ -13,25 +13,128 @@
  * limitations under the License.
  */
 
-#if defined(__LINUX__)
-
 #include "iotjs_def.h"
 #include "iotjs_objectwrap.h"
 #include "iotjs_module_gpioctl.h"
 
+//
+// this is dummy code for x86.
+// please add new 'device' file for your purpose.
+// use timer with 1msec delay to implement async callback.
+// real gpio implementation would be better to use threading
+//
 
 namespace iotjs {
 
-/*
- * Use default dummy for linux-x86
- */
+
+// type for delayed callback
+typedef enum {
+  DELAYEDCALL_SETSPIN = 1,
+} dc_type;
+
+
+// base class for delayed callback
+struct DelayedCallBase {
+  dc_type type;
+  uv_timer_t timer;
+};
+
+
+// setspin delayed callback class
+struct DelayedCallSetSpin : public DelayedCallBase {
+  gpio_setpin_t* data;
+  gpio_setpin_cb cb;
+  int res;
+};
+
+
+static void timerHandleTimeout(uv_timer_t* handle) {
+  uv_timer_stop(handle);
+
+  DelayedCallBase* base;
+  base = reinterpret_cast<DelayedCallBase*>(handle->data);
+  switch (base->type) {
+    case DELAYEDCALL_SETSPIN : {
+      DelayedCallSetSpin* psetspin =
+                            reinterpret_cast<DelayedCallSetSpin*>(base);
+      psetspin->cb(reinterpret_cast<gpio_cb_t*>(psetspin->data),
+                   psetspin->res);
+      break;
+    }
+  }
+}
+
+
+//-----------------------------------------------------------------------------
+
+class GpioControlImpl : public GpioControl {
+public:
+  explicit GpioControlImpl(JObject& jgpioctl);
+
+  virtual int Initialize(void);
+  virtual void Release(void);
+  virtual int SetPin(gpio_setpin_t* setpin, gpio_setpin_cb cb);
+  virtual int SetPin(int32_t pin, int32_t dir, int32_t mode);
+};
+
 
 GpioControl* GpioControl::Create(JObject& jgpioctl)
 {
-  return new GpioControl(jgpioctl);
+  return new GpioControlImpl(jgpioctl);
 }
 
 
-} // namespace iotjs
+GpioControlImpl::GpioControlImpl(JObject& jgpioctl)
+    : GpioControl(jgpioctl) {
+}
+
 
-#endif // __NUTTX__
+int GpioControlImpl::Initialize(void) {
+  if (_fd > 0 )
+    return GPIO_ERR_INITALIZE;
+
+  DDDLOG("x86 linux gpio dummy initalize");
+  _fd = 1;
+  return _fd;
+
+}
+
+
+void GpioControlImpl::Release(void) {
+  DDDLOG("x86 linux gpio dummy release");
+  _fd = 0;
+}
+
+
+// SetPin for callback is provided
+int GpioControlImpl::SetPin(gpio_setpin_t* setpin_data, gpio_setpin_cb cb) {
+  DDDLOG("x86 linux gpio dummy setpin w/cb pin(%d), dir(%d), mode(%d)",
+         setpin_data->pin, setpin_data->dir, setpin_data->mode);
+
+  Environment* env = Environment::GetEnv();
+  DelayedCallSetSpin* delay = new DelayedCallSetSpin;
```

Comment:
Why do we have to this delayed callback here? even this function is dummy so didn't actually set pin configuration.